packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,278 @@
+   *
+   * @default - None
+   */
+  readonly labels?: { [name: string]: string };
+  /**
+   * The IAM role to associate with your node group. The Amazon EKS worker node kubelet daemon
+   * makes calls to AWS APIs on your behalf. Worker nodes receive permissions for these API calls through an IAM instance profile
+   * and associated policies. Before you can launch worker nodes and register them into a cluster, you must create an IAM role for
+   * those worker nodes to use when they are launched.
+   *
+   * @default - None. Auto-generated if not specified.
+   */
+  readonly nodeRole?: IRole;
+  /**
+   * The AMI version of the Amazon EKS-optimized AMI to use with your node group (for example, `1.14.7-YYYYMMDD`).
+   *
+   * @default - The latest available AMI version for the node group's current Kubernetes version is used.
+   */
+  readonly releaseVersion?: string;
+  /**
+   * The remote access (SSH) configuration to use with your node group. Disabled by default, however, if you specify an Amazon EC2
+   * SSH key but do not specify a source security group when you create a managed node group, then port 22 on the worker nodes is
+   * opened to the internet (0.0.0.0/0)
+   *
+   * @default - disabled
+   */
+  readonly remoteAccess?: NodegroupRemoteAccess;
+  /**
+   * The metadata to apply to the node group to assist with categorization and organization. Each tag consists of a key and an optional
+   * value, both of which you define. Node group tags do not propagate to any other resources associated with the node group, such as the
+   * Amazon EC2 instances or subnets.
+   *
+   * @default - None
+   */
+  readonly tags?: { [name: string]: string };
+}
+
+/**
+ * NodeGroup properties interface
+ */
+export interface NodegroupProps extends NodegroupOptions {
+  /**
+   * Cluster resource
+   * [disable-awslint:ref-via-interface]"
+   */
+  readonly cluster: Cluster;
+}
+
+/**
+ * The Nodegroup resource class
+ */
+export class Nodegroup extends Resource implements INodegroup {
+  /**
+   * Import the Nodegroup from attributes
+   */
+  public static fromNodegroupName(scope: Construct, id: string, nodegroupName: string): INodegroup {
+    class Import extends Resource implements INodegroup {
+      public readonly nodegroupName = nodegroupName;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * ARN of the nodegroup
+   *
+   * @attribute
+   */
+  public readonly nodegroupArn: string;
+  /**
+   * Nodegroup name
+   *
+   * @attribute
+   */
+  public readonly nodegroupName: string;
+  /**
+   * the Amazon EKS cluster resource
+   *
+   * @attribute ClusterName
+   */
+  public readonly cluster: Cluster;
+  /**
+   * IAM role of the instance profile for the nodegroup
+   */
+  public readonly role: IRole;
+
+  private readonly desiredSize: number;
+  private readonly maxSize: number;
+  private readonly minSize: number;
+
+  constructor(scope: Construct, id: string, props: NodegroupProps ) {
+    super(scope, id);
```

Comment:
We have a pattern around physical names. See an example in e.g. s3.Bucket. 

Pass physicalName to “super” and use super.getResourceName and getResourceArn.